### DISCUSSION TITLE
What scale is WiFi.RSSI() in?

### DISCUSSION THREAD

**Discussion by the-maldridge** [earlephilhower/arduino-pico#1927] [Q&A] [answered]
I'm trying to get from this scale to a dBm value as observed by the Pico for the currently connected access point while in station mode.  I am trying to work out what the scale factor is between the uint8 that gets handed back, and the scale I would expect wifi signal strength to be measured in.  The comments in the cyw43 driver imply this is some value just getting returned from the firmware on the chip, but without further context its not clear what the scale that's using either.  Any advice is appreciated.

**Comment by earlephilhower** [comment, marked as answer]
That's a good question and not documented anywhere that I could find.  I assumed it was RSSI in dBm already, since it gets less negative the more powerful the WiFi signal, and the scale seemed right (minimum RSSI of ~-75, and RSSI of ~-20 for right next to my AP).  You might as at the [driver's author's repo](https://github.com/georgerobotics/cyw43-driver) and see if he has an answer.

Alternatively, I believe the same chip is supported by the Linux CYW driver, so you could take a look-see there.

**Reply by the-maldridge** [reply]
So chasing this down through the drivers I think ultimately this was an error in my code and I think this is actually in dBm.  Your comment that you were getting negative numbers made me think something was off with my values since I was getting number around 170-220.  While the driver internally moves things around as `uint8`, by the time the value has been Arduino-fied it comes back as a signed `int`.  Swapping the types out made the numbers look correct as reasonably dBm values.
